### PR TITLE
Add PreCommit Java IO AWS job

### PR DESCRIPTION
**Pull request by Amar3tto**
The following jenkins jobs have been migrated:

https://ci-beam.apache.org/job/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct_Commit/
https://ci-beam.apache.org/job/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct_Cron/
https://ci-beam.apache.org/job/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct_Phrase/

### How has it been tested?
The job has been tested on Self-hosted GKE runners ubuntu 20.04.
The following test cases are passed:

- by creating pull requests with changes in trigger paths
- by adding a comment to the PR
- by making a push

Screenshot from 2023-08-04 16-16-14

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: .github/workflows/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Rerun on comment
+        if: github.event.comment.body == 'Run Java_Amazon-Web-Services_IO_Direct PreCommit'
+        uses: ./.github/actions/rerun-job-action
+        with:
+          pull_request_url: ${{ github.event.issue.pull_request.url }}
+          github_repository: ${{ github.repository }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          github_job: ${{ github.job }}
+          github_current_run_id: ${{ github.run_id }}
```
Comment (andreydevyatkin) [inline .github/workflows/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct.yml]: Please update these steps in accordance with recent changes:

```
- uses: actions/checkout@v3
- name: Setup repository
  uses: ./.github/actions/setup-action
  with:
    comment_phrase: 'Run Java_Amazon-Web-Services_IO_Direct PreCommit'
    github_token: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+jobs:
+  beam_PreCommit_Java_Amazon-Web-Services_IO_Direct:
+    name: beam_PreCommit_Java_Amazon-Web-Services_IO_Direct (Run Java_Amazon-Web-Services_IO_Direct PreCommit)
+    timeout-minutes: 60
+    if: |
+      github.event_name == 'push' || 
+      github.event_name == 'pull_request_target' || 
+      github.event.comment.body == 'Run Java_Amazon-Web-Services_IO_Direct PreCommit' || 
+      github.event_name == 'schedule'
```
Comment (andreydevyatkin) [inline .github/workflows/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct.yml]: Please add the condition:
```
github.event_name == 'workflow_dispatch' ||
```

**Review by andreydevyatkin — CHANGES_REQUESTED** [review]
see the inline comments

File: .github/workflows/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct.yml (hunk 1 above)
Comment (Amar3tto) [reply .github/workflows/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct.yml]: Done

File: .github/workflows/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct.yml (hunk 2 above)
Comment (Amar3tto) [reply .github/workflows/beam_PreCommit_Java_Amazon-Web-Services_IO_Direct.yml]: Done

**Review by andreydevyatkin — APPROVED** [review]
LGTM

**Comment by mosche** [comment]
Just mentioning, this module is deprecated and flagged for removal some time soon. The relevant replacement is amazon-web-services2.

**Review by Abacn — APPROVED** [review]

**Pull request merged**